src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1740,6 +1762,8 @@ impl Cage {
                     drop(pardir_inodeobj);
                     log_metadata(&FS_METADATA, parent_inodenum);       
                 }
+                NET_METADATA.domsock_paths.remove(&true_oldpath);
+                NET_METADATA.domsock_paths.insert(true_newpath);
```

Comment:
@jesings we possibly could also fix this by just swapping the order right?

Reply:
If the insert is guaranteed to fail when some other process already inserted the new path, then this swapping would make sense. Otherwise, we do need to add some locking here.

Reply:
https://docs.rs/dashmap/latest/dashmap/struct.DashSet.html#method.insert

since the hashset only contains unique elements it should fail if it tries to double insert.

Reply:
I also don't believe that these are used in any way where a race condition would be harmful.